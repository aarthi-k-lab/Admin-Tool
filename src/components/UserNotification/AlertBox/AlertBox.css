.alert-box {
  align-items: center;
  border-radius: 0.2rem;
  display: flex;
  justify-content: flex-start;
  padding: 0.5rem 0.5rem;
  width: 100%;
}

.alert-box--error {
  composes: alert-box;
  background-color: var(--red-50);
  border: solid 0.125rem var(--red-600);
}

.alert-box--success {
  composes: alert-box;
  background-color: var(--light-green-50);
  border: solid 0.125rem var(--light-green-500);
}

.error-icon.error-icon {
  color: var(--red-a700);
}

.success-icon.success-icon {
  color: var(--light-green-a700);
}

.message {
  padding-left: 0.5rem;
}
