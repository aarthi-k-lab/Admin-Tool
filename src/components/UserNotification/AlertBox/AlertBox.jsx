import React from 'react';
import PropTypes from 'prop-types';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import WarningIcon from '@material-ui/icons/Warning';
import './AlertBox.css';

class AlertBox extends React.PureComponent {
  static getIcon(level) {
    switch (level) {
      case this.ERROR:
        return <WarningIcon styleName="error-icon" />;
      case this.SUCCESS:
        return <CheckCircleIcon styleName="success-icon" />;
      default:
        return null;
    }
  }

  static getBoxStyle(level) {
    switch (level) {
      case this.ERROR:
        return 'alert-box--error';
      case this.SUCCESS:
        return 'alert-box--success';
      default:
        return 'alert-box';
    }
  }

  render() {
    const { level, message } = this.props;
    const icon = this.constructor.getIcon(level);
    return (
      <div styleName={this.constructor.getBoxStyle(level)}>
        {icon}
        <span styleName="message">{message}</span>
      </div>
    );
  }
}

AlertBox.ERROR = 'error';
AlertBox.SUCCESS = 'success';

AlertBox.propTypes = {
  level: PropTypes.oneOf([AlertBox.ERROR, AlertBox.SUCCESS]).isRequired,
  message: PropTypes.string.isRequired,
};

export default AlertBox;
