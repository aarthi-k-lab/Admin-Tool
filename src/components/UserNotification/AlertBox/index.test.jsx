import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AlertBox from '.';

describe('<AlertBox />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AlertBox level="error" message="Please retry" />);
  });

  it('snapshot test for error level', () => {
    const tree = renderer
      .create(<AlertBox level="error" message="Please retry" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot test for success level', () => {
    const tree = renderer
      .create(<AlertBox level="success" message="Enjoy your cupcake!" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot test for other levels', () => {
    const tree = renderer
      .create(<AlertBox level="info" message="Informational message" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the AlertBox component', () => {
    expect(wrapper.find('span').text()).toBe('Please retry');
  });
});
