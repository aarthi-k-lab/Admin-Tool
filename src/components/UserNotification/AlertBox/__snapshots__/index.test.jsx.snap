// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertBox /> snapshot test for error level 1`] = `
<div
  className="components-UserNotification-AlertBox-AlertBox__alert-box--error--1aOFR components-UserNotification-AlertBox-AlertBox__alert-box--cNmYG"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1 components-UserNotification-AlertBox-AlertBox__error-icon--1Nupt"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
    />
  </svg>
  <span
    className="components-UserNotification-AlertBox-AlertBox__message--J2alF"
  >
    Please retry
  </span>
</div>
`;

exports[`<AlertBox /> snapshot test for other levels 1`] = `
<div
  className="components-UserNotification-AlertBox-AlertBox__alert-box--cNmYG"
>
  <span
    className="components-UserNotification-AlertBox-AlertBox__message--J2alF"
  >
    Informational message
  </span>
</div>
`;

exports[`<AlertBox /> snapshot test for success level 1`] = `
<div
  className="components-UserNotification-AlertBox-AlertBox__alert-box--success--24cUu components-UserNotification-AlertBox-AlertBox__alert-box--cNmYG"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1 components-UserNotification-AlertBox-AlertBox__success-icon--ySD2I"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
  </svg>
  <span
    className="components-UserNotification-AlertBox-AlertBox__message--J2alF"
  >
    Enjoy your cupcake!
  </span>
</div>
`;
